private/src/gst/gstavbvideosrc.cpp

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright (C) 2018 Intel Corporation. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <gst/gst.h>
+#include <gst/base/gstbasesrc.h>
+#include "gst/gstavbvideosrc.h"
+#include "gst/gstavbclock.h"
+
+#include <arpa/inet.h>
+#include <sys/types.h>
+#include <sys/syscall.h>
+#include <unistd.h>
+
+#include <cstdio>
+
+GST_DEBUG_CATEGORY_STATIC(gst_avbvideosrc_debug_category);
+#define GST_CAT_DEFAULT gst_avbvideosrc_debug_category
+
+#define DEFAULT_STREAM_NAME "media_transport.avb.mpegts_streaming.7"
+
+/* prototypes */
+
+static GstClock *gst_avbvideosrc_provide_clock(GstElement *element);
+static void gst_avbvideosrc_finalize(GObject * object);
+
+static GstFlowReturn gst_avbvideosrc_create(GstPushSrc * src, GstBuffer ** buf);
+
+static void mpegts_callback(ias_avbvideobridge_receiver *receiver, bool sph, ias_avbvideobridge_buffer const * packet, void *user_ptr);
+
+static void gst_avbvideosrc_set_property(GObject *object, guint prop_id,
+        const GValue *value, GParamSpec *pspec);
+static void gst_avbvideosrc_get_property(GObject *object, guint prop_id,
+        GValue *value, GParamSpec *pspec);
+static GstStateChangeReturn gst_avbvideosrc_change_state(GstElement *element,
+        GstStateChange transition);
+
+enum
+{
+    PROP_0,
+    PROP_STREAM_NAME
+};
+
+/* pad templates */
+
+static GstStaticPadTemplate gst_avbvideosrc_src_template =
+GST_STATIC_PAD_TEMPLATE("src",
+        GST_PAD_SRC,
+        GST_PAD_ALWAYS,
+        GST_STATIC_CAPS(
+            "video/mpegts,"
+            "  packetsize = (int) 188,"
```

Comment:
Please add a `#define` for 188. I am seeing it repeated in a few places.

Reply:
Ok. But here I would need a second define as string, or some magic preprocessing stuff, like two levels of macros.

Reply:
(Or be a rebel and let here as is. `C`, why won't you help me?)

Reply:
Another solution would be to build that capabilities string more dynamically, is there a not too awkward way of doing that?

Reply:
Oh, https://developer.gnome.org/glib/stable/glib-Miscellaneous-Macros.html#G-STRINGIFY:CAPS came to the rescue =D

Reply:
I was thinking about not using GST_STATIC_CAPS and using something else. But this is getting too complicated, I agree. I think the two `#defines` (one for the value, the other for the string) are fine for now. What do you think?